golangci.yml

```diff
@@ -0,0 +1,2310 @@
+    # Default: false
+    explicit-exhaustive-map: true
+
+  exhaustivestruct:
+    # Struct Patterns is list of expressions to match struct packages and names.
+    # The struct packages have the form `example.com/package.ExampleStruct`.
+    # The matching patterns can use matching syntax from https://pkg.go.dev/path#Match.
+    # If this list is empty, all structs are tested.
+    # Default: []
+    struct-patterns:
+      - '*.Test'
+      - 'example.com/package.ExampleStruct'
+
+  exhaustruct:
+    # List of regular expressions to match struct packages and names.
+    # If this list is empty, all structs are tested.
+    # Default: []
+    include:
+      - '.*\.Test'
+      - 'example\.com/package\.ExampleStruct[\d]{1,2}'
+    # List of regular expressions to exclude struct packages and names from check.
+    # Default: []
+    exclude:
+      - 'cobra\.Command$'
+
+  forbidigo:
+    # Forbid the following identifiers (list of regexp).
+    # Default: ["^(fmt\\.Print(|f|ln)|print|println)$"]
+    forbid:
+      - ^print.*$
+      - 'fmt\.Print.*'
+      # Optionally put comments at the end of the regex, surrounded by `(# )?`
+      # Escape any special characters.
+      - 'fmt\.Print.*(# Do not commit print statements\.)?'
+    # Exclude godoc examples from forbidigo checks.
+    # Default: true
+    exclude_godoc_examples: false
+
+  funlen:
+    # Checks the number of lines in a function.
+    # If lower than 0, disable the check.
+    # Default: 60
+    lines: -1
+    # Checks the number of statements in a function.
+    # If lower than 0, disable the check.
+    # Default: 40
+    statements: -1
+
+  gci:
+    # DEPRECATED: use `sections` and `prefix(github.com/org/project)` instead.
+    local-prefixes: github.com/org/project
+
+    # Section configuration to compare against.
+    # Section names are case-insensitive and may contain parameters in ().
+    # The default order of sections is `standard > default > custom > blank > dot`,
+    # If `custom-order` is `true`, it follows the order of `sections` option.
+    # Default: ["standard", "default"]
+    sections:
+      - standard                       # Standard section: captures all standard packages.
+      - default                        # Default section: contains all imports that could not be matched to another section type.
+      - prefix(github.com/org/project) # Custom section: groups all imports with the specified Prefix.
+      - blank                          # Blank section: contains all blank imports. This section is not present unless explicitly enabled.
+      - dot                            # Dot section: contains all dot imports. This section is not present unless explicitly enabled.
+
+    # Skip generated files.
+    # Default: true
+    skip-generated: false
+
+    # Enable custom order of sections.
+    # If `true`, make the section order the same as the order of `sections`.
+    # Default: false
+    custom-order: true
+
+  gocognit:
```

Comment:
do we benefit from having both this *and* the cyclomatic thing? my default preference would be "neither" really.

Reply:
as we're starting this I'd rather maximize signal, and deal with noise later.
Though to your previous comments, I updated this to its default of 30"